src/screens/WaitTimeScreen.tsx

```diff
@@ -22,103 +22,148 @@ import {
   }
 }
 
 export const WaitTimeScreen = ({ navigation }: IWaitTimeScreenProps) => {
   const { theme } = useContext(ThemeContext);
+  const { user } = useContext(AuthContext);
+
   const [searchValue, setSearchValue] = useState("");
   const [isVisible, setIsVisible] = useState(false);
   const [sortBy, setSortBy] = useState(SortByEnum.DISTANCE);
   const [refreshing, setRefreshing] = useState(false);
+  const [rawPOIData, setRawPOIData] = useState<POI[]>([]);
+  const [poiFilters, setPOIFilters] = useState<Set<POIType>>(new Set());
+  const [aggregateData, setAggregateData] = useState(DEFAULT_AGGREGATE_DATA);
 
   let searchBar: SearchBar | null;
 
-  const poiTypes = ["Food", "Transport", "Gym", "Library"];
-  const steps1 = [
-    { title: "5 mins", description: "Lowest" },
-    { title: "7 mins", description: "Average" },
-    { title: "17 mins", description: "Highest" },
+  const filters = [
+    { display: "Food", val: POIType.EATERY },
+    { display: "Shop", val: POIType.SHOP },
+    { display: "Transport", val: POIType.TRANSPORT },
+    { display: "Gym", val: POIType.GYM },
+    { display: "Library", val: POIType.LIBRARY },
   ];
 
-  const poiData = [
-    {
-      waitTime: 10,
-      location: "Centro",
-      distance: 0.5,
-      lastUpdated: 5,
-      type: "food",
-    },
-    {
-      waitTime: 7,
-      location: "MUSC",
-      distance: 0.6,
-      lastUpdated: 8,
-      type: "shopping",
-    },
-    {
-      waitTime: 15,
-      location: "Location 3",
-      distance: 1.2,
-      lastUpdated: 0,
-      type: "food",
-    },
-    {
-      waitTime: 4,
-      location: "Location 4",
-      distance: 1.5,
-      lastUpdated: 68,
-      type: "transport",
-    },
-    {
-      waitTime: 6,
-      location: "Location 5",
-      distance: 0.2,
-      lastUpdated: 1,
-      type: "shopping",
-    },
-    {
-      waitTime: 17,
-      location: "Location 6",
-      distance: 0.1,
-      lastUpdated: 4,
-      type: "food",
-    },
-    {
-      waitTime: 4,
-      location: "Location 7",
-      distance: 2.0,
-      lastUpdated: 0,
-      type: "food",
-    },
-  ];
+  const poiData = rawPOIData.filter((poi) => {
+    // Filter by search
+    return (
+      // No search or type filters
+      (searchValue.length === 0 && poiFilters.size === 0) ||
+      // Filter by search
+      (searchValue.length > 0 &&
+        poi.name.toLowerCase().includes(searchValue.toLowerCase())) ||
+      // Filter by type
+      (poiFilters.size > 0 && poiFilters.has(POIType[poi.type as POIType]))
+    );
+  });
 
-  const onRefresh = useCallback(() => {
-    setRefreshing(true);
-    setTimeout(() => {
-      setRefreshing(false);
-    }, 1000);
+  /**
+   * Fetch POIs from the API
+   */
+  const fetchPOIs = async () => {
+    try {
+      const res = await poiApi.getAllPOI(
+        43.262983, // TODO: replace with user's LATITUDE
+        -79.918611, // TODO: replace with user's LONGITUDE
```

Comment:
Imma merge my PR as soon as I test it with my phone, then you can replace this with the location state

Reply:
Done and dusted.